packages/dds/tree/src/util/rangeMap.ts

```diff
@@ -3,25 +3,30 @@
  * Licensed under the MIT License.
  */
 
-import { oob } from "@fluidframework/core-utils/internal";
+import { newTupleBTree, type TupleBTree } from "./bTreeUtils.js";
 
 /**
  * RangeMap represents a mapping from integers to values of type T or undefined.
  * The values for a range of consecutive keys can be changed or queried in a single operation.
  */
 export class RangeMap<T> {
-	private readonly entries: RangeEntry<T>[];
+	private readonly tree: TupleBTree<[number], RangeEntry<T>>;
 
-	public constructor(initialEntries?: RangeEntry<T>[]) {
-		this.entries = initialEntries ? [...initialEntries] : [];
+	public constructor() {
+		this.tree = newTupleBTree();
 	}
 
 	/**
 	 * Retrieves all entries from the rangeMap.
 	 * @returns An array of RangeEntryResult objects, each containing the start index, length, and value of a contiguous range.
```

Comment:
```suggestion
	 * Retrieves all entries from the rangeMap.
```

Reply:
I didn't think that the @return comment had a lot of value: it't not saying anything that the signature doesn't convey.